### PR TITLE
make jxlinfo output nicer for humans

### PR DESCRIPTION
**Pull request by jonsneyers**
Currently, `jxlinfo` produces rather verbose output that is somewhat hard to understand for anyone not very familiar with libjxl.

This makes the output a lot more compact and nicer to read for humans. "Unsurprising" information like e.g. intrinsic dimensions that are equal to the image dimensions is dropped, leaving only the info that is actually interesting to see.

Layers are now also shown (not just animation frames), including crop dimensions/offsets.

Some example outputs:

Before:
```
dimensions: 960x860
have_container: 0
uses_original_profile: 0
bits_per_sample: 8
have_preview: 0
have_animation: 0
intrinsic xsize: 960
intrinsic ysize: 860
orientation: 1 (Normal)
num_color_channels: 3
num_extra_channels: 1
extra channel 0:
  type: Alpha
  bits_per_sample: 8
  alpha_premultiplied: 0 (Non-premultiplied)
color profile:
  format: JPEG XL encoded color profile
  color_space: 0 (RGB color)
  white_point: 1 (D65)
  primaries: 1 (sRGB)
  transfer_function: 13 (sRGB)
  rendering_intent: 1 (Relative)
frame:
  still frame, unnamed
```

After:
```
JPEG XL image, 960x860, lossy, 8-bit RGBA
Color space: RGB, D65, sRGB primaries, sRGB transfer function, rendering intent: Relative
```

Before:
```
box: type: "JXL " size: 12
box: type: "ftyp" size: 20
box: type: "Exif" size: 366
box: type: "xml " size: 30261
box: type: "jbrd" size: 883
box: type: "jxlc" size: 72798
dimensions: 1000x500
have_container: 1
uses_original_profile: 1
bits_per_sample: 8
have_preview: 0
have_animation: 0
intrinsic xsize: 1000
intrinsic ysize: 500
orientation: 1 (Normal)
num_color_channels: 3
num_extra_channels: 0
color profile:
  format: JPEG XL encoded color profile
  color_space: 0 (RGB color)
  white_point: 1 (D65)
  primaries: 1 (sRGB)
  transfer_function: 13 (sRGB)
  rendering_intent: 1 (Relative)
frame:
  still frame, unnamed
```

After:
```
JPEG XL file format container (ISO/IEC 18181-2)
Uncompressed Exif metadata: 366 bytes
Uncompressed xml  metadata: 30261 bytes
JPEG bitstream reconstruction data available
JPEG XL image, 1000x500, (possibly) lossless, 8-bit RGB
Color space: RGB, D65, sRGB primaries, sRGB transfer function, rendering intent: Relative
```

Before:
```
dimensions: 307x230
have_container: 0
uses_original_profile: 1
bits_per_sample: 8
have_preview: 0
have_animation: 1
ticks per second (numerator / denominator): 100 / 1
num_loops: 0
have_timecodes: 0
intrinsic xsize: 307
intrinsic ysize: 230
orientation: 1 (Normal)
num_color_channels: 3
num_extra_channels: 1
extra channel 0:
  type: Alpha
  bits_per_sample: 8
  alpha_premultiplied: 0 (Non-premultiplied)
color profile:
  format: JPEG XL encoded color profile
  color_space: 0 (RGB color)
  white_point: 1 (D65)
  primaries: 1 (sRGB)
  transfer_function: 13 (sRGB)
  rendering_intent: 0 (Perceptual)
frame:
  duration: 8 ticks (80.000000 ms)
frame:
  duration: 8 ticks (80.000000 ms)
frame:
  duration: 8 ticks (80.000000 ms)
frame:
  duration: 8 ticks (80.000000 ms)
frame:
  duration: 8 ticks (80.000000 ms)
frame:
  duration: 8 ticks (80.000000 ms)
frame:
  duration: 8 ticks (80.000000 ms)
frame:
  duration: 8 ticks (80.000000 ms)
```

After:
```
JPEG XL animation, 307x230, (possibly) lossless, 8-bit RGBA
Color space: RGB, D65, sRGB primaries, sRGB transfer function, rendering intent: Perceptual
frame: full image size, duration: 80.0 ms
frame: 277x186 at position (5,39), duration: 80.0 ms
frame: 282x180 at position (0,40), duration: 80.0 ms
frame: 285x183 at position (0,40), duration: 80.0 ms
frame: 278x183 at position (7,40), duration: 80.0 ms
frame: 295x185 at position (7,39), duration: 80.0 ms
frame: 286x185 at position (16,39), duration: 80.0 ms
frame: 274x186 at position (9,39), duration: 80.0 ms
Animation length: 0.640 seconds (looping)
```

Before:
```
box: type: "JXL " size: 12
box: type: "ftyp" size: 20
box: type: "jxll" size: 9
box: type: "jxlp" size: 5430004
dimensions: 3072x3072
have_container: 1
uses_original_profile: 1
bits_per_sample: 8
have_preview: 0
have_animation: 0
intrinsic xsize: 3072
intrinsic ysize: 3072
orientation: 1 (Normal)
num_color_channels: 3
num_extra_channels: 2
extra channel 0:
  type: K (of CMYK)
  bits_per_sample: 8
extra channel 1:
  type: Alpha
  bits_per_sample: 8
  alpha_premultiplied: 0 (Non-premultiplied)
color profile:
  format: ICC profile
  ICC profile size: 557168
  CMM type: "ADBE"
  color space: "CMYK"
  rendering intent: 0
box: type: "jxlp" size: 4931453
frame:
  still frame, unnamed
box: type: "brob" size: 39133
```

After:
```
JPEG XL file format container (ISO/IEC 18181-2)
JPEG XL image, 3072x3072, (possibly) lossless, 8-bit CMYK+Alpha
Color space: 557168-byte ICC profile, CMM type: "ADBE", color space: "CMYK", rendering intent: 0
layer: 2214x2420 at position (399,265)
layer: full image size
Brotli-compressed psdr metadata: 39133 compressed bytes
```

### CODE REVIEW

File: lib/include/jxl/decode.h [hunk 1]
```diff
@@ -1199,8 +1199,8 @@ JXL_EXPORT JxlDecoderStatus JxlDecoderSetDecompressBoxes(JxlDecoder* dec,
  *
  * @param dec decoder object
  * @param type buffer to copy the type into
- * @param decompressed which box type to get: JXL_TRUE to get the raw box type,
- * which can be "brob", JXL_FALSE, get the underlying box type.
+ * @param decompressed which box type to get: JXL_FALSE to get the raw box type,
```
Comment (veluca93) [inline lib/include/jxl/decode.h:1202]: Can you move this to a separate PR? Seems fairly unrelated.

File: tools/jxlinfo.c [hunk 2]
```diff
@@ -79,109 +82,161 @@ int PrintBasicInfo(FILE* file) {
...
+             info.xsize, info.ysize);
+      if (info.uses_original_profile)
+        printf("(possibly) lossless");
+      else
+        printf("lossy");
+      printf(", %d-bit ", info.bits_per_sample);
       if (info.exponent_bits_per_sample)
-        printf("float, with exponent_bits_per_sample: %d\n",
-               info.exponent_bits_per_sample);
+        printf("float (%d exponent bits) ", info.exponent_bits_per_sample);
```
Comment (veluca93) [inline tools/jxlinfo.c:93]: Or any other statement :)

File: tools/jxlinfo.c [hunk 3]
```diff
@@ -13,7 +13,7 @@
 
 #include "jxl/decode.h"
 
-int PrintBasicInfo(FILE* file) {
+int PrintBasicInfo(FILE* file, int verbose) {
   uint8_t* data = NULL;
   size_t data_size = 0;
   // In how large chunks to read from the file and try decoding the basic info.
```
Comment (veluca93) [inline tools/jxlinfo.c:19]: If we're going to install this as an utility, I would perhaps have somewhat bigger chunks than 64 bytes :)

File: lib/include/jxl/decode.h (hunk 1 above)
Comment (jonsneyers) [reply lib/include/jxl/decode.h:1202]: Right, it's 'related' because I found this documentation bug while working on jxlinfo, but it's indeed better in a separate PR. See https://github.com/libjxl/libjxl/pull/1363

File: tools/jxlinfo.c (hunk 3 above)
Comment (jonsneyers) [reply tools/jxlinfo.c:19]: Bumped it up to 2 kb.

File: tools/jxlinfo.c [hunk 4]
```diff
@@ -79,109 +82,160 @@ int PrintBasicInfo(FILE* file) {
...
+        printf("Grayscale");
+      else {
+        if (cmyk) {
+          printf("CMYK");
+          cmyk = 0;
+        } else if (alpha) {
+          printf("RGBA");
+          alpha = 0;
+        } else
+          printf("RGB");
```
[the remark this answers is not in the document]
Comment (jonsneyers) [reply tools/jxlinfo.c:116]: I do this too often — can we somehow make clang-format do this automatically?

I have a tendency to write stuff like
```
if (foo) bar;
else baz;
```

which to me looks fine, but then clang-format rewrites it to four lines but doesn't add braces...

[the remark this answers is not in the document]
Comment (veluca93) [reply tools/jxlinfo.c:116]: Apparently it needs either clang-tidy or clang-format version 15+...

https://stackoverflow.com/questions/26111162/can-clang-format-add-braces-to-single-line-if-statements-etc

[the remark this answers is not in the document]
Comment (jonsneyers) [reply tools/jxlinfo.c:116]: I guess I'll wait then until my clang-format is at that version...

[the remark this answers is not in the document]
Comment (jonsneyers) [reply tools/jxlinfo.c:116]: I _think_ I added {} to all of these, there were a couple more.

**Review by veluca93 — APPROVED** [review]

**Review by veluca93 — APPROVED** [review]
